File: sanidad-backend/src/main/java/com/farmacia/sanidadbackend/dto/DevolucionResponse.java
package com.farmacia.sanidadbackend.dto;

import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

@Data
public class DevolucionResponse {

    private Long id;
    private String numeroDevolucion;
    private Long ventaId;
    private String numeroFactura;

    private Long usuarioSolicitanteId;
    private String usuarioSolicitanteNombre;

    private Long usuarioApruebaId;
    private String usuarioApruebaNombre;

    private String estado; // PENDIENTE, APROBADA, RECHAZADA
    private String motivo;
    private LocalDateTime fechaSolicitud;
    private LocalDateTime fechaAprobacion;

    private BigDecimal subtotalDevuelto;
    private BigDecimal ivaDevuelto;
    private BigDecimal totalDevuelto;

    private BigDecimal montoDevueltoEfectivo;
    private BigDecimal montoDevueltoSaldo;

    private List<DevolucionDetalleResponse> detalles;
}

File: SanidadMovil/app/src/main/java/com/sanidad/movil/data/remote/dto/DevolucionResponse.kt
package com.sanidad.movil.data.remote.dto

data class DevolucionResponse(
    val id: Long,
    val numeroDevolucion: String,
    val ventaId: Long,
    val numeroFactura: String,
    val usuarioSolicitanteId: Long,
    val usuarioSolicitanteNombre: String,
    val usuarioApruebaId: Long?,
    val usuarioApruebaNombre: String?,
    val estado: String,
    val motivo: String?,
    val fechaSolicitud: String,
    val fechaAprobacion: String?,
    val subtotalDevuelto: Double,
    val ivaDevuelto: Double,
    val totalDevuelto: Double,
    val montoDevueltoEfectivo: Double,
    val montoDevueltoSaldo: Double,
    val detalles: List<DevolucionDetalleResponse>
)